### DISCUSSION TITLE
creating a snippect collection on github or bitbucket - how to achive that ?

### DISCUSSION THREAD

**Discussion by fsj-digital** [community/community#22349] [New to GitHub]
<p>hello dear all,</p>
<p>i am pretty new to github and bitbucket.</p>
<p>on  bitbucket i want to create a repo  containing a list of several  /(approx  15 to 20 ) different files that contain</p>
<p>code for learning… so that it looks like so:</p>
<p>i want to get such  stucture: <a href="https://github.com/CoreyMSchafer/code_snippets" rel="nofollow noopener">https://github.com/CoreyMSchafer/code_snippets</a></p>
<p>code_snippet1</p>
<p>code_snippet2</p>
<p>code_snippet3</p>
<p>code_snippet4</p>
<p>code_snippet5</p>
<p>code_snippet6</p>
<p>code_snippet7</p>
<p>code_snippet8</p>
<p>code_snippet9</p>
<p>code_snippet10</p>
<p>and so forth. .</p>
<p>again: i want to get such  stucture: <a href="https://github.com/CoreyMSchafer/code_snippets" rel="nofollow noopener">https://github.com/CoreyMSchafer/code_snippets</a></p>
<p>well - how can i do that - how can i achieve that!?</p>

**Comment by Wabri** [comment, marked as answer]
<p>In order to do what you ask you need to follow this algorithm:</p>
<ol>
<li>
<strong>Create</strong> repository on bitbucket</li>
<li>
<strong>Clone</strong> the repository in local</li>
<li>Inside the repository create directory called " <strong>code_snippet1</strong>" and put inside of it all the file relative to the <strong>snippet1</strong>
</li>
<li>Than create directory called " <strong>code_snippet2</strong>" and put inside of it all the file relative to the <strong>snippet2</strong>
</li>
<li>Than create directory called " <strong>code_snippet3</strong>" and put inside of it all the file relative to the <strong>snippet3</strong>
</li>
<li><strong>And so on…</strong></li>
<li>Now you need to <strong>add, commit and push</strong> this files, this depends on how you want to do this:</li>
<li>With commands:<br>
1.</li>
</ol>
<pre><code>git add *

2. 

git commit -m "create snippets"

3. 

git push origin master
</code></pre>
<ol start="2">
<li>With a graphical interface:<br>
1. <a href="https://desktop.github.com/" rel="nofollow noopener">Github desktop</a><br>
2. <a href="https://www.gitkraken.com/" rel="nofollow noopener">Gitkraken</a><br>
3. Other git interface: <a href="https://alternativeto.net/software/github-desktop/" rel="nofollow noopener">similar github desktop apps</a>
</li>
<li>if you need to add some more snippets <strong>return to 3</strong>
</li>
<li>Else you have <strong>done</strong>
</li>
</ol>
<p>Ask for more details…</p>
<p>-Gabriele-</p>
